app.Makefile

```diff
@@ -0,0 +1,131 @@
+ifndef __APP_MAKEFILE__
```

Comment:
I see all these Makefiles are from the GCP project, and were previously included directly from the vendored directory. 

Do we need all of them? I see only three directly in use in the makefile. Do some depend on the others?

Additionally can we take them out of the root and put them in a subfolder? That would make their difference compared to the actual makefile a bit clearer.

Finally how can we update these files? What is the procedure? What version are these from? Is that what the `MARKETPLACE_TOOLS_TAG` is for? Can we turn that procedure into a simple Makefile command? 

In any case, please document how you have upgraded the marketplace tools so it may be repeated easily.

Reply:
Added a readme, moved files to `tools/`, and removed files that weren't needed!